### ISSUE TITLE
build.sh script does not work on aarch64

### ISSUE THREAD

**Issue by llelectronics** [AppImage/AppImageKit#563]
Build.sh script is using **uname -p** to determine the arch which prints out **unknown** on aarch64. 
So I suggest using **uname -m** instead.

**Comment by TheAssassin** [comment]
Good idea. Please provide details on where you tested the behavior.
If you send a PR, I'll merge it as soon as possible.

**Comment by llelectronics** [comment]
Tested it on a Pinebook with Ubuntu Mate and Debian installed.

**Comment by probonopd** [comment]
I think we had some back and forth on this earlier (and I had assumed it was fixed back then - suprising how hard it is to get something as "simple" as this get right on Linux), e.g., here:

* https://github.com/AppImage/AppImageKit/issues/91#issuecomment-189697757
* https://github.com/AppImage/AppImageKit/issues/325
* https://github.com/AppImage/AppImageKit/issues/456
* https://github.com/AppImage/AppImageKit/issues/312
* https://github.com/AppImage/AppImageKit/pull/427

> as says man uname, uname -p is not portable, replaced with uname -m

So yes, if you still encounter this, please send a PR, thanks.

**Comment by notklaatu** [comment]
For what it's worth, over at SlackBuilds.org we use ``uname -m`` with success. 

That said, we officially support only i?86 and x86_64 so possibly some varieties of ARM are getting misidentified.   THAT said, I've never had a problem with it on any of my ARM devices.

tl;dr: +1 for ``uname -m``

**Comment by TheAssassin** [comment]
Has either of you ever tried the `arch` command? It's part of the `coreutils`.

**Comment by TheAssassin** [comment]
@llelectronics turned out this issue is a bit more complex than we assumed at first. I ran a simple `grep -R "uname -p"` on my workstation, and could observe that not only our tools are using `uname -p` to detect the architecture, but also most of the dependencies do the same.

What platform was the one which you were testing this initially?

The information on your use case is missing, unfortunately, however I think it should suffice to change our code, though, as I could imagine that the problems arise when building the AppImages, whose filenames miss the architecture, right?

**Comment by llelectronics** [comment]
The plattform I tried it on was an arm64 Pinebook with an Allwinner chip in it.(I think this was clear from the title and comments)

**Comment by TheAssassin** [comment]
What OS did you run there?

**Comment by llelectronics** [comment]
Debian and Ubuntu

**Comment by TheAssassin** [comment]
@llelectronics mind to reply to my other question?

At which point did the usage of `uname -p` start to break things, e.g., the tools' AppImages filenames, or something like that?

**Comment by llelectronics** [comment]
Only the script. I don't know about other parts really.

**Comment by TheAssassin** [comment]
The issue is a little outdated, we don't have an architecture detection any more in `build.sh` (this file is inheriting the environment variable from the Travis CI infrastructure, otherwise local builds should rather use CMake directly). I replaced it in all other locations, though.

**Comment by llelectronics** [comment]
Okay then this issue can be closed I guess.

**Comment by TheAssassin** [comment]
Will be auto-closed with the merge of #650.

**Comment by krayon** [comment]
@TheAssassin @notklaatu : like `uname -p`, `arch` also isn't portable, though
mainly as it's not installed by default as part of `coreutils` (
https://www.gnu.org/software/coreutils/manual/html_node/arch-invocation.html ).
